Document External SQL Replication (#1530)
platformsh/platformsh-docs: docs/src/tutorials/mysql-replication.md

```bash
platform sql -r replication
```

Now you need to prevent any changes to the data while you view the binary log position. You'll use this to tell the replica at exactly which point it should start replicating from. On the primary server, flush and lock all tables by running `FLUSH TABLES WITH READ LOCK`. Keep this session running - exiting it will release the lock. Get the current position in the binary log by running `SHOW MASTER STATUS`:

```sql
mysql> FLUSH TABLES WITH READ LOCK;
Query OK, 0 rows affected (0.016 sec)

mysql> SHOW MASTER STATUS;
+-----------------+----------+--------------+------------------+
| File            | Position | Binlog_Do_DB | Binlog_Ignore_DB |
+-----------------+----------+--------------+------------------+
| binlogs.000002  |  1036    | dflt         |                  |
+-----------------+----------+--------------+------------------+
```

Record the File and Position details. If binary logging has just been enabled, these will be blank. Now, with the lock still in place, copy the data from the primary to the replica.

Login to the app container, then run:

```sh

> # Ddump the data from primary. Note that it will dump only the databases, which "replicator" user has access to

Reviewer: ```suggestion
# Dump the data from primary. Note that it will dump only the databases, which "replicator" user has access to.
```

$ mysqldump --all-databases --single-transaction -h database.internal -P 3306 -u replicator -p > /path/to/dump.sql
```

Download the dump file, then move it to the server where your replica lives to import it.

```bash
# Copy the dump to your replica
$ mysql -u root < /path/to/dump.sql
```

Note for live databases: You just need to make a local copy of the data, you don't need to keep the primary locked until the replica has imported the data.  Once the `mysqldump` has completed, you can release the lock on the primary by running `UNLOCK TABLES`.

```sql
mysql> UNLOCK TABLES;
```

## Setting up the replica

### Configuring the replica

> As mentioned above you have to set up a replica on your own.  We assume that you have a running MariaDB/MySQL replica instance,  give the replica a unique `server_id` (distinct from primary). You can find out primary's `server_id` by running:

Reviewer: ```suggestion
As mentioned above you have to set up a replica on your own. Assuming that you have a running MariaDB/MySQL replica instance, give the replica a unique `server_id` (distinct from primary). You can find out primary's `server_id` by running:
```

```sql
mysql> SHOW VARIABLES LIKE 'server_id';

+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| server_id     | 1     |
+---------------+-------+
```

> Then set a distinct `server_id` number (e.g. server_id+1) in you replica config (e.g. `my.cnf`) under:

Reviewer: ```suggestion
Then set a distinct `server_id` number (e.g. server_id+1) in your replica config (e.g. `my.cnf`) under:
```
